enterprise/server/api/api_server.go

```diff
@@ -526,14 +526,18 @@ func (s *APIServer) ExecuteWorkflow(ctx context.Context, req *apipb.ExecuteWorkf
 	requestCtx := requestcontext.ProtoRequestContextFromContext(ctx)
 
 	wfID := wfs.GetLegacyWorkflowIDForGitRepository(user.GetGroupID(), req.GetRepoUrl())
+	branchName := req.GetReportingMetadata().GetBranch()
+	if branchName == "" {
+		branchName = req.GetRef()
```

Comment:
It seems like the branch name could wind up being a commit SHA here if the user forgets to pass `branch` in the metadata - will that break anything on the CI runner?

Reply:
That will just mean the metadata in the UI will show that the branch name is the commit SHA, which is not ideal and may be confusing, but shouldn't break anything else